### PR TITLE
fix: always keep date picker calendar popup inside of viewport

### PR DESCRIPTION
**Pull request by sashuk**
[FX-4445]

### Description

This pull request makes DatePicker popper fit the **viewport** in the most extreme cases when the screen is small, so overlapping the input is allowed ([confirmed with Design Team in Slack](https://toptal-core.slack.com/archives/CERF5NHT3/p1698319400470039?thread_ts=1697120927.351179&cid=CERF5NHT3))

### How to test

- [Temploy](https://picasso.toptal.net/FX-NULL-fit-date-picker)
- [Storybook] Navigate to https://picasso.toptal.net/FX-NULL-fit-date-picker/?path=/story/overlays-modal--modal and trigger the date picker from the `Default` case modal on different screen sizes
- [Talent Portal] Follow the steps
  - check out branch from https://github.com/toptal/talent-portal-frontend/pull/5535
  - run `yarn install`
  - log in as https://staff-portal.toptal.net/talents/4220731
  - run `yarn start:staging`
  - navigate to https://local-development.staging.toptal.net:8443/portal/payments and click `Download payment` button to open the modal

https://github.com/toptal/picasso/assets/1390758/6bfcadad-831e-4941-8427-713d7012960d

### Screenshots

https://github.com/toptal/picasso/assets/1390758/e5e64eec-0ff8-4e18-b2c6-66eb195207e1

### Development checks

- [x] Add changeset according to [guidelines](https://github.com/toptal/picasso/blob/master/docs/contribution/changeset-guidelines.md) (if needed)
- [x] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/contribution/component-api.md)
- [x] Make sure that additions and changes on the design follow [Toptal's BASE design](https://design.toptal.net/), and it's been already discussed with designers at #-base-core
- [N/A] Annotate all `props` in component with documentation
- [N/A] Create `examples` for component
- [x] Ensure that deployed demo has expected results and good examples
- [x] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).
- [x] Self reviewed
- [N/A] Covered with tests ([visual tests](https://github.com/toptal/picasso/blob/master/docs/contribution/visual-testing.md) included)
- [x] ⚠️ **Revert changes made in Storybook before the merge**

> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

[FX-4445]: https://toptal-core.atlassian.net/browse/FX-4445?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by sashuk** [comment]
@toptal-bot run package:alpha-release

**Comment by sashuk** [comment]
@toptal-bot run package:alpha-release

File: packages/picasso/src/DatePicker/constants.ts [hunk 1]
```diff
@@ -1,12 +1,15 @@
+import type { PopperOptions } from 'popper.js'
+
 export const DEFAULT_DATE_PICKER_EDIT_DATE_FORMAT = 'MM-dd-yyyy'
 export const DEFAULT_DATE_PICKER_DISPLAY_DATE_FORMAT = 'MMM d, yyyy'
-export const DEFAULT_POPPER_OPTIONS = {
+export const POPPER_OPTIONS: PopperOptions = {
```
Comment (sashuk) [inline packages/picasso/src/DatePicker/constants.ts:5]: No need in `DEFAULT`, as it is not changes anywhere else

File: packages/picasso/src/DatePicker/constants.ts [hunk 2]
```diff
@@ -1,12 +1,15 @@
...
   modifiers: {
     hide: {
       enabled: false,
     },
     preventOverflow: {
-      enabled: false,
+      enabled: true,
+      boundariesElement: 'viewport',
```
Comment (sashuk) [inline packages/picasso/src/DatePicker/constants.ts:12]: More details in https://popper.js.org/docs/v1/#modifiers..preventOverflow.boundariesElement

File: packages/picasso/src/Modal/story/Default.example.tsx [hunk 3]
```diff
@@ -1,49 +1,21 @@
...
-    text: 'Alabama',
-    value: 'Alabama',
-  },
-  {
-    text: 'Utah',
-    value: 'Utah',
-  },
-]
-
+// TODO: revert to the original file before the merge
```
Comment (sashuk) [inline packages/picasso/src/Modal/story/Default.example.tsx:5]: There is an check list item in the pull request description not to forget to do it

**Comment by sashuk** [comment]
@toptal-bot run package:alpha-release

File: .changeset/two-foxes-type.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+---
+'@toptal/picasso': minor
```
Comment (sashuk) [inline .changeset/two-foxes-type.md:2]: Not a breaking change, as it improves the current behavior in very rare cases

**Comment by sashuk** [comment]
@toptal-anvil ping reviewers

File: .changeset/two-foxes-type.md (hunk 4 above)
Comment (TomasSlama) [reply .changeset/two-foxes-type.md:2]: might be even patch, as we just fix styling, not really new feature 🤔 WDYT?

**Review by TomasSlama — APPROVED** [review]
good job! Looks good

**Review by mkrl — APPROVED** [review]
Nice job! Works exactly as expected, tested in temploy.

File: .changeset/two-foxes-type.md (hunk 4 above)
Comment (sashuk) [reply .changeset/two-foxes-type.md]: @TomasSlama Agree, updated!

**Pull request merged**